### PR TITLE
Feat/vite build

### PR DESCRIPTION
**Pull request by chmelevskij**
Web based version build of configurator. Not ready for prod but can live alongside nwjs.

Mainly done:
* Add [vite](https://vitejs.dev/) based setup for build tooling.
* Make both nwjs and vite versions work along each other.
* Basic web based serial implementation.

To run vite build `yarn dev`

### CODE REVIEW

File: src/js/serial_backend.js
Comment (haslinghuis) [inline src/js/serial_backend.js:15]: :heart:

File: src/js/webSerial.js
Comment (haslinghuis) [inline src/js/webSerial.js:43]: https://github.com/betaflight/betaflight-configurator/blob/55fabf2c49737bc54099407f3fe56913bfa6d681/src/js/serial.js#L25-L32

Comment (haslinghuis) [reply src/js/webSerial.js:43]: We use the filtering to only show and use relevant devices

File: package.json
Comment (chmelevskij) [inline package.json:142]: had to pin this to newer version as node-gyp was failing on macos arm

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/msp.js
Comment (haslinghuis) [inline src/js/msp.js:310]:
```suggestion
        const connected = import.meta.env ? serial.connected : !!serial.connectionId;
```

Comment (chmelevskij) [reply src/js/msp.js:310]: Looking at the code now, it should probably be 
```suggestion
        const connected = import.meta.env ? serial.connected : !serial.connectionId;
```

Comment (chmelevskij) [reply src/js/msp.js:310]: Hmm, actually, is should probably be `!`
```suggestion
        const connected = import.meta.env ? serial.connected : !serial.connectionId;
```

**Comment by nerdCopter** [comment]
`5765510b`

```
yarn install --check-files
yarn dev   # can it be `yarn vite`?
```

Findings:
- setup tab & receiver tab: no quadcopter rendered.
- all tabs with `save` button seem to work.
- all tabs with `save & reboot` button: save and reboot properly triggers, but UI does not disconnect (remains on open tab);  pressing `Disconnect` does not recover, requires refresh/reload web-page.
- presets tab: no sources/settings available.
- PIDs tab: Mode `RPY` does not display full set of sliders.
- motors tab mixer preview images broken.
- OSD blank.
- CLI tab broken, no input/output.

**Comment by chmelevskij** [comment]
P.S. could someone check the old version as well as new?

**Comment by nerdCopter** [comment]
> P.S. could someone check the old version as well as new?

Do you mean older firmware?  Or do you mean standalone `yarn gulp [release|debug]`?

**Comment by haslinghuis** [comment]
> Findings:

setup tab & receiver tab: no quadcopter rendered.
all tabs with save button seem to work.
all tabs with save & reboot button: save and reboot properly triggers, but UI does not disconnect (remains on open tab); pressing Disconnect does not recover, requires refresh/reload web-page.
presets tab: no sources/settings available.
PIDs tab: Mode RPY does not display full set of sliders.
motors tab: mixer preview images broken.
OSD tab: empty/blank.
CLI tab: broken, no input/output.

None of these issues are present with KAKUTEH7

image

**Comment by nerdCopter** [comment]
> None of these issues are present with KAKUTEH7

with the `yarn dev` (http://localhost:5173/# webUI)?
i even get a string version 0.0.1
image

**Comment by haslinghuis** [comment]
Y or N <enter> does not continue here?

image

**Comment by nerdCopter** [comment]
> Y or N does not continue here?

i had the same. it's rhetorical, you can still go to the URL.  eventually it went away for me. no idea why.

**Review by nerdCopter — APPROVED** [review]

**Comment by chmelevskij** [comment]
> Y or N does not continue here?

Yeah, it's warning. it never should hit nwjs branches anyway so should be fine. Till we get rid of it 😈

**Comment by chmelevskij** [comment]
@nerdCopter nwjs was as chrome as this web serial requirement. You still can use Chromium as well if you prefer no google in browser.

**Pull request merged**
